### PR TITLE
Integrate namespace in SafeMath

### LINKED ISSUE #307: Integrate namespace in SafeMath

### PR DESCRIPTION
**Pull request by andrew-fleming**
Resolves #307.

### CODE REVIEW

File: src/openzeppelin/security/safemath.cairo [hunk 1]
```diff
@@ -7,109 +7,114 @@ from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
...
-    uint256_check(a)
-    uint256_check(b)
-    let (c: Uint256, is_overflow) = uint256_add(a, b)
-    with_attr error_message("Safemath: addition overflow"):
-        assert is_overflow = FALSE
+namespace SafeMath:
+
+    # Adds two integers. 
+    # Reverts if the sum overflows.
+    func uint256_checked_add{
```
Comment (martriay) [inline src/openzeppelin/security/safemath.cairo:18]:
```suggestion
    func uint256_add{
```
now that there's a namespace I think we should drop some redundant info from the name. we could even drop uint256 if we call it SafeUint256 or something like that

File: src/openzeppelin/security/safemath.cairo [hunk 2]
```diff
@@ -7,109 +7,114 @@ from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
...
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        } (a: Uint256, b: Uint256) -> (c: Uint256):
+        uint256_check(a)
+        uint256_check(b)
+        let (c: Uint256, is_overflow) = uint256_add(a, b)
+        with_attr error_message("Safemath: addition overflow"):
```
Comment (martriay) [inline src/openzeppelin/security/safemath.cairo:26]:
```suggestion
        with_attr error_message("SafeMath: addition overflow"):
```
this is the spelling everywhere else

**Review by martriay — CHANGES_REQUESTED** [review]
Looks good! I suggested a function renaming. Why the rename of the test contract functions though?

File: src/openzeppelin/security/safemath.cairo (hunk 1 above)
Comment (andrew-fleming) [reply src/openzeppelin/security/safemath.cairo:18]: I originally switched it to `uint256_add` (and the rest), but it conflicts with the Cairo Uint256 imports which is why I kept the long names. Changing the dir/namespace to SafeUint256 is a good idea and should fix that

File: src/openzeppelin/security/safemath.cairo (hunk 2 above)
Comment (andrew-fleming) [reply src/openzeppelin/security/safemath.cairo:26]: On it!

**Comment by andrew-fleming** [comment]
> Looks good! I suggested a function renaming. Why the rename of the test contract functions though?

I think using the actual method name is easier to follow especially when calling it in the tests `contract.uint256_add()` vs `contract.test_add()`. I can switch it back if you prefer the other way

**Comment by martriay** [comment]
No strong opinions

**Review by martriay — APPROVED** [review]

**Pull request merged**